Sources/Stevenson/CircleCIService.swift

```diff
@@ -51,4 +55,90 @@ public struct CircleCIService {
             }
         }
     }
+
+}
+
+extension CircleCIService {
+
+    public struct PipelineRequest: Encodable {
+        let branch: String
+        let parameters: [String: Parameter]
+
+        public enum Parameter: Encodable {
+            case bool(Bool)
+            case string(String)
+
+            public func encode(to encoder: Encoder) throws {
+                var values = encoder.singleValueContainer()
+                switch self {
+                case let .bool(value):
+                    try values.encode(value)
+                case let .string(value):
+                    try values.encode(value)
+                }
+            }
+        }
+    }
+
+    public struct PipelineResponse: Content {
+        public let branch: String
+        public let buildURL: URL
+    }
+
+    struct PipelineID: Content {
+        let id: String
+    }
+
+    struct Pipeline: Content {
+        let workflows: [Workflow]
+        let vcs: VCS
+
+        struct Workflow: Content {
+            let id: String
+        }
+
+        struct VCS: Content {
+            let branch: String
+        }
+    }
+
+    private func pipelineURL(pipelineID: String) -> URL {
+        return URL(
+            string: "/api/v2/pipeline/\(pipelineID)?circle-token=\(token)",
+            relativeTo: baseURL
+        )!
+    }
```

Comment:
Maybe move that as a function on `struct PipelineID` instead?

Reply:
I'd keep this separate from data models, with separate mapping of responses it looks a bit nicer now